x-pack/plugins/security/server/lib/authorization/saved_objects/ensure_saved_objects_privileges.test.ts

```diff
@@ -0,0 +1,405 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Actions } from '..';
+import { ensureSavedObjectsPrivilegesFactory } from '.';
+import { DEFAULT_SPACE_ID } from '../../../../../spaces/common/constants';
+
+const createMockErrors = () => {
+  const forbiddenError = new Error('Mock ForbiddenError');
+  const generalError = new Error('Mock GeneralError');
+
+  return {
+    forbiddenError,
+    decorateForbiddenError: jest.fn().mockReturnValue(forbiddenError),
+    generalError,
+    decorateGeneralError: jest.fn().mockReturnValue(generalError),
+  };
+};
+
+const createMockAuditLogger = () => {
+  return {
+    savedObjectsAuthorizationFailure: jest.fn(),
+    savedObjectsAuthorizationSuccess: jest.fn(),
+  };
+};
+
+const createMockActions = () => {
+  return {
+    savedObject: {
+      get(type: string, action: string) {
+        return `mock-saved_object:${type}/${action}`;
+      },
+    },
+  } as Actions;
+};
+
+const createCheckSavedObjectsPrivileges = (
+  spacesEnabled: boolean,
+  checkPrivilegesImpl: () => Promise<any>
+) => {
+  const mockAuditLogger = createMockAuditLogger();
+  const mockErrors = createMockErrors();
+  const mockActions = createMockActions();
+
+  const checkSavedObjectsPrivileges = jest.fn(
+    ensureSavedObjectsPrivilegesFactory({
+      actionsService: mockActions,
+      auditLogger: mockAuditLogger,
+      checkPrivilegesWithRequest: () => ({
```

Comment:
The tests don't ensure that `checkPrivilegesWithRequest` is called with the actual request itself. We could potentially pass `checkPrivileges` itself to the `ensureSavedObjectsPrivilegesFactory` instead of `checkPrivilegesWithRequest`, since we only use `request` for this within the wrapper.